### ISSUE TITLE
s390x fails with -gsplit-dwarf

### ISSUE THREAD

**Issue by okias** [rui314/mold#940]
```
[109/1366] Compiling C++ object src/gtest/libgtest.a.p/src_gtest-all.cc.o
[110/1366] Linking static target src/gtest/libgtest.a
[111/1366] Linking target src/util/xmlconfig_test
FAILED: src/util/xmlconfig_test 
/usr/lib/ccache/s390x-linux-gnu-g++  -o src/util/xmlconfig_test src/util/xmlconfig_test.p/tests_xmlconfig.cpp.o -Wl,--as-needed -Wl,--no-undefined -Wl,--gdb-index -Wl,--start-group src/util/libmesa_util.a src/util/format/libmesa_format.a src/util/libmesa_util_sse41.a src/c11/impl/libmesa_util_c11.a src/util/libxmlconfig.a src/gtest/libgtest.a /usr/lib/s390x-linux-gnu/libz.so -pthread -lm -ldl /usr/lib/s390x-linux-gnu/libexpat.so -Wl,--end-group
collect2: fatal error: ld terminated with signal 11 [Segmentation fault], core dumped
```
CI job: https://gitlab.freedesktop.org/mesa/mesa/-/jobs/34008841

Do you want additional information?

P.S. it's mold `1.6.0` or `1.8.0` with applied [Rename inttypes.h integers.h](https://github.com/rui314/mold/commit/34bfbc0983e478e081a7023e61138d0baf801fec) . That's what you get when you fix issues, more issues @rui314 :rofl:

**Comment by rui314** [comment]
Let me build mesa on s390 on Docker to reproduce the issue.

**Comment by rui314** [comment]
It might be related to `split-dwarf`. What if you disable that option?

**Comment by okias** [comment]
You have been right. It's the `-gsplit-dwarf`.

**Comment by rui314** [comment]
Can you try again with the above commit?

**Comment by okias** [comment]
Confirming it fixed the issue (running clean 1.9.0). Thank you, Rui!
